[WIP] syncs i18n documentation with SecureDrop Client

cfm opened the pull request:
## Status

**Work in progress** as part of freedomofpress/securedrop-client#1304

## Description of Changes

TK

## Testing

TK

## Release 

TK

## Checklist (Optional)

- [ ] Doc linting (`make docs-lint`) passed locally
- [ ] Doc link linting (`make docs-linkcheck`) passed
- [ ] You have previewed (`make docs`) docs at http://localhost:8000

creviera commented:
Planning to review this today given xtra free time :-)

creviera commented on docs/development/translations.rst:
```diff
@@ -22,6 +28,13 @@ about anything in this document, here's how to ask for help:
 * Chat in the `SecureDrop instant messaging channel`_
+
+   * Localization Lab, with whom we :ref:`coordinate SecureDrop's
+     translation<when_is_securedrop_translated>`, also maintains `their own
```
I think this link is a bit confusing because it takes you to _when_ translations can happen, which felt more low level than how we coordinate with Localization Lab. Also this is in the context of "Getting help" so I think pointing to their channel and who they are would be enough for this section.

cfm replied:
Fair!  Simplified in 6ac2ad8d6c26c0dd6edd7e69cfe066bef85fa588.

creviera commented on docs/development/translations.rst:
```diff
@@ -22,6 +28,13 @@ about anything in this document, here's how to ask for help:
 * Post a message in the `translations category of the SecureDrop forum`_
 * Chat in the `SecureDrop instant messaging channel`_
+
+   * Localization Lab, with whom we :ref:`coordinate SecureDrop's
```
A link to https://www.localizationlab.org/ would be better here than later on in the doc when we're getting into the weeds of when translations happen during code changes and such.

cfm replied:
Added in 6ac2ad8d6c26c0dd6edd7e69cfe066bef85fa588!

creviera commented on docs/development/translations.rst:
```diff
@@ -96,6 +115,25 @@ SecureDrop is translated using the **Weblate** platform. :ref:`Get started using
+the release process.
+
+Both processes are coordinated in collaboration with `Localization Lab
+<https://www.localizationlab.org/>`_.  You can watch for the announcements
```
Helpful link to https://www.localizationlab.org/ but should move up to the top with the first mention of the org.

cfm replied:
As of 6ac2ad8d6c26c0dd6edd7e69cfe066bef85fa588, kept both links for good measure, understanding that folks may enter the documentation at different points.

creviera reviewed (changes requested):
content lgtm and links worked

cfm commented:
Thanks for reviewing, @creviera!  I believe I've addressed your feedback with 6ac2ad8d6c26c0dd6edd7e69cfe066bef85fa588, which I'm open to squashing in if you agree.

creviera reviewed (approved):
lgtm
